src/ch/ch_conf.c

```diff
@@ -171,6 +171,62 @@ virCHDriverConfigDispose(void *obj)
     g_free(cfg->logDir);
 }
 
+int
+virCHVersionString(const char *str, unsigned long *version,
+                      bool allowMissing)
+{
+    unsigned int major, minor = 0, micro = 0;
+
+    if (strncmp(str, "cloud-hypervisor v", 18) == 0) {
+        /**
+         * Check for regular 'cloud-hypervisor --version' output with major,minor and micro version
+         * This format is for upstream CLH release version
+         * For ex: cloud-hypervisor v32.0.0
+         */
+        if(sscanf(str, "cloud-hypervisor v%d.%d.%d", &major, &minor, &micro) != 3){
+            /**
+             * Try extracting major and minor version only
+             * This format is for upstream CLH main branch build
+             * For ex: cloud-hypervisor v33.0-104-ge0e3779e-dirty
+             */
+            if(sscanf(str, "cloud-hypervisor v%d.%d", &major, &minor) != 2){
+
+                /**
+                 * Log Error if none of the format matches and return -1
+                 */
+                fprintf(stderr, "Can not extract upstream CLH version: %s\n", str);
+                return -1;
+            }
+        }
+    } else {
+        /**
+         * Check for msft 'cloud-hypervisor --version' output
+         * This format is for msft CLH main branch build
+         * For ex: cloud-hypervisor msft/v32.0.131-1-ga5d6db5c-dirty
+         */
+        if(sscanf(str, "cloud-hypervisor msft/v%d.%d.%d", &major, &minor, &micro) != 3){
```

Comment:
This seems specific to our internal implementation. I'd suggest making this version extraction as generic as possible.

I can think of following sequence of steps:
1) Chop off all version info until the last '/', if one exists. At this point the version could be one of the following formats (with all branch prefixes chopped off):
     v32.0.0  or v32.0.131-1-ga5d6db5c

2) Once you just have version info, follow above logic to extract the version information.

This will allow us to rename or added more prefixes to branch if needed, without any code changes.

Reply:
@praveen-pk  Done in latest push. Please review.